src/Avalonia.Base/Utilities/StringTokenizer.cs

```diff
@@ -70,8 +70,15 @@ public int ReadInt32(char? separator = null)
         public bool TryReadDouble(out double result, char? separator = null)
         {
             var success = TryReadString(out var stringResult, separator);
-            result = success ? double.Parse(stringResult, _formatProvider) : 0;
-            return success;
+
+            if (success)
+            {
+                success = double.TryParse(stringResult, NumberStyles.Float, _formatProvider, out result);
```

Comment:
I think this could be written more simply as:

```
if (TryReadString(out var stringResult, separator) &&
    double.TryParse(stringResult, NumberStyles.Float, _formatProvider, out result))
```